src/Common/src/TypeSystem/Interop/IL/MarshalHelpers.cs

```diff
@@ -59,6 +59,22 @@ public static bool IsBlittableType(TypeDesc type)
             return false;
         }
 
+        public static bool IsStructMarshallingRequired(TypeDesc typeDesc)
+        {
+            if (typeDesc.Category != TypeFlags.ValueType)
+                return false;
+
+
+            var type = typeDesc.UnderlyingType as MetadataType;
```

Comment:
Seems like this code doesn't handle anything but `typeDesc` being a `MetadataType`. Maybe make it part of the contract by enforcing it in the signature?